First draft (#639)
OctopusDeploy/blog: blog/2022-07/gitlab-external-feeds/index.md

---
title: Using GitLab feeds with Octopus Deploy
description: Learn how to connect GitLab feed types to Octopus Deploy.
author: shawn.sesna@octopus.com
visibility: public 
published: 2022-07-20-1400
metaImage: blogimage-usinggitlabfeedswithoctopus-2022.png
bannerImage: blogimage-usinggitlabfeedswithoctopus-2022.png
bannerImageAlt: Packages on a conveyer belt start in an orange machine, travel through a blue machine, and come out deployed with version numbers.
isFeatured: false
tags:
 - DevOps
 - Continuous Integration
---

> Octopus Deploy contains a [built-in repository](https://octopus.com/docs/packaging-applications/package-repositories/built-in-repository) that supports a variety of [file types](https://octopus.com/docs/packaging-applications#supported-formats).  However, some customers prefer to use a third-party repository or use the repositories built in to their Continuous Integration (CI) tools.  

Reviewer: ```suggestion
Octopus Deploy has a [built-in repository](https://octopus.com/docs/packaging-applications/package-repositories/built-in-repository) that supports a variety of [file types](https://octopus.com/docs/packaging-applications#supported-formats).  However, some customers prefer to use a third-party repository or use the repositories built in to their Continuous Integration (CI) tools.  
```
```suggestion
Octopus Deploy contains a [built-in repository](https://octopus.com/docs/packaging-applications/package-repositories/built-in-repository) that supports a variety of [file types](https://octopus.com/docs/packaging-applications#supported-formats).  However, some customers prefer to use a third-party repository or use the repositories built in to their Continuous Integration (CI) tools.  
```

> In this post, I demonstrate how to use the registries built in to GitLab as external feeds for Octopus Deploy.

Reviewer: ```suggestion
In this post, I demonstrate how to use GitLab's built-in registries as external feeds to deploy your projects with Octopus.
```

- (Optional) **Credentials**: Username and password to access the feed

![Octopus dashboard showing Create Feeds screen under External Feeds section](octopus-add-nuget-feed.png)

Click **SAVE AND TEST** to ensure the feed is functional.  Enter the name of the package (can be partial):

![Octopus External Feed](octopus-test-nuget-feed.png)

### Adding a GitLab container feed

Unlike the other 2 feed types, the container feed doesn't need the Project or Group ID.  It does, however, need to know which port on the server is configured to host the container registry.

Fill in the form fields:

- **Feed Type**: `Docker Container Registry`
- **Name**: Give the feed a name, for example, `GitLab Container Registry`
- **URL**: `https://[gitlabserver]:[port]`
- **Credentials**: Username and password to access the feed

![Octopus dashboard showing Create Feeds screen under External Feeds section](octopus-add-container-feed.png)

Click **SAVE AND TEST** to ensure the feed is functional. Enter the name of the package (can be partial):

![Octopus External Feed](octopus-test-container-feed.png)

## Conclusion

> In many cases, you'll benefit from using Octopus Deploy's built-in repository. However, if you need something cross-space and don't want to duplicate, there are alternatives. This post showed you how to use the built-in registries of GitLab as external feeds to Octopus Deploy.  

Reviewer: ```suggestion
In many cases, you'll benefit from using Octopus Deploy's built-in repository. However, if you need something cross-space and don't want to duplicate, there are alternatives. This post showed you how to use GitLab's built-in registries as external feeds to deploy your projects with Octopus.
```
